lib/fluent/plugin_helper/server.rb

```diff
@@ -709,13 +709,15 @@ def try_tls_accept
                 return true
               end
             rescue Errno::EPIPE, Errno::ECONNRESET, Errno::ETIMEDOUT, Errno::ECONNREFUSED, Errno::EHOSTUNREACH => e
+              peeraddr = (@_handler_socket.peeraddr rescue PEERADDR_FAILED)
               @log.trace "unexpected error before accepting TLS connection",
```

Comment:
I may be wrong, I thought that `@_handler_socket.peeraddr` (derived from  OpenSSL::SSL::SocketForwarder#peeraddr) will raise Errno::EXXX. 
Will `rescue PEERADDR_FAILED` work as expectedly?

Reply:
(I know this idiom `rescue PEERADDR_FAILED` is used in other code, but I'm not sure it can be used here, too)

Reply:
Hmm, `#peeraddr` and `PEERADDR_FAILED` returned array format is same but I'm not aware of exceptions from `@_handler_socket.peeraddr`.
OpenSSL's `peeraddr` is defined in https://github.com/ruby/openssl/blob/master/lib/openssl/ssl.rb#L246.
And this methods behavior is same for [IPSocket#peeraddr](https://docs.ruby-lang.org/ja/latest/method/IPSocket/i/peeraddr.html):
https://docs.ruby-lang.org/ja/latest/method/OpenSSL=3a=3aSSL=3a=3aSocketForwarder/i/peeraddr.html

Reply:
Yeah, we should catch `Errno::EXXX` exceptions:
https://pubs.opengroup.org/onlinepubs/009695399/functions/getpeername.html
This exceptions should be from getpeername(2):
https://pubs.opengroup.org/onlinepubs/009695399/functions/getpeername.html

Reply:
Sorry, I don't understand what do you worry.
Do you worry whether `foo rescue bar` idiom works also against `Errno::EXXX`?
I confirmed that it works as expected:
```ruby
irb(main):001:0> PEERADDR_FAILED = ["?", "?", "name resolution failed", "?"]
=> ["?", "?", "name resolution failed", "?"]
irb(main):002:1* def peeraddr
irb(main):003:1*   raise Errno::ENOTCONN
irb(main):004:0> end
=> :peeraddr
irb(main):005:0> a = (peeraddr rescue PEERADDR_FAILED)
=> ["?", "?", "name resolution failed", "?"]
irb(main):006:0> b = peeraddr
Traceback (most recent call last):
        5: from /usr/bin/irb:23:in `<main>'
        4: from /usr/bin/irb:23:in `load'
        3: from /usr/lib/ruby/gems/2.7.0/gems/irb-1.2.6/exe/irb:11:in `<top (required)>'
        2: from (irb):6
        1: from (irb):3:in `peeraddr'
Errno::ENOTCONN (Transport endpoint is not connected)
```

Reply:
Thanks, surely it works as expected. I've misunderstood `oo rescue bar` behavior.

Reply:
Thanks. Errno exceptions inherit `StandardError`. So, it works as expected.
https://github.com/fluent/fluentd/pull/3484